packages/react-aria-components/docs/FileTrigger.mdx

```diff
@@ -0,0 +1,208 @@
+import {Keyboard} from '@react-spectrum/text';
+import Button from '@react-spectrum/docs/pages/assets/component-illustrations/ActionButton.svg';
+import Link from '@react-spectrum/docs/pages/assets/component-illustrations/Link.svg';
+
+---
+category: Drag and drop
+keywords: [filetrigger, dropzone, aria, accessibility]
+type: component
+---
+
+# FileTrigger
+
+<PageDescription>{docs.exports.FileTrigger.description}</PageDescription>
+
+<HeaderInfo
+  packageData={packageData}
+  componentNames={['FileTrigger']} />
+
+## Example
+
+```tsx example
+import {FileTrigger} from 'react-aria-components';
+import {Button} from 'react-aria-components';
+
+<FileTrigger>
+  <Button>Select a file</Button>
+</FileTrigger>
+```
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
+
+```css
+.react-aria-Button {
+  --border-color: var(--spectrum-alias-border-color);
+  --border-color-pressed: var(--spectrum-alias-border-color-down);
+  --border-color-disabled: var(--spectrum-alias-border-color-disabled);
+  --background-color: var(--spectrum-global-color-gray-50);
+  --background-color-pressed: var(--spectrum-global-color-gray-100);
+  --text-color: var(--spectrum-alias-text-color);
+  --text-color-disabled: var(--spectrum-alias-text-color-disabled);
+  --focus-ring-color: slateblue;
+
+  color: var(--text-color);
+  background: var(--background-color);
+  border: 1px solid var(--border-color);
+  border-radius: 4px;
+  appearance: none;
+  vertical-align: middle;
+  font-size: 1.2rem;
+  text-align: center;
+  margin: 0;
+  outline: none;
+  padding: 4px 12px;
+
+  &[data-pressed] {
+    box-shadow: inset 0 1px 2px rgb(0 0 0 / 0.1);
+    background: var(--background-color-pressed);
+    border-color: var(--border-color-pressed);
+  }
+
+  &[data-focus-visible] {
+    border-color: var(--focus-ring-color);
+    box-shadow: 0 0 0 1px var(--focus-ring-color);
+  }
+
+  &:disabled {
+    border-color: var(--border-color-disabled);
+    color: var(--text-color-disabled);
+  }
+}
+
+@media (forced-colors: active) {
+  .react-aria-Button {
+    forced-color-adjust: none;
+    --border-color: ButtonBorder;
+    --border-color-pressed: ButtonBorder;
+    --border-color-disabled: GrayText;
+    --background-color: ButtonFace;
+    --background-color-pressed: ButtonFace;
+    --text-color: ButtonText;
+    --text-color-disabled: GrayText;
+    --focus-ring-color: Highlight;
+  }
+}
+```
+
+</details>
+
+## Features
+
+Access to a computer’s file system can be created in HTML using an `<input>` with a file `type` attribute. However, file input's on their own are not very useful since they only allow users to select files from their devices. As a result, file inputs are commonly used inside a `<form>` alongside  a `<button>` to submit the form. `FileTrigger` simplifies this by setting up the input element's configuration with reference to a pressable component, such as a button, for a customizable file upload. 
```

Comment:
I think we should focus this on allowing custom styling. FileTrigger is basically just a hidden `<input type="file">` that is triggered by a custom button. Maybe this could be a starting point:

"A file input can be created with an `<input type="file">` element, but this supports very limited styling options. `FileTrigger` works with a pressable child such as a `Button` to create accessible file inputs that can be styled as needed."